test/test_poll.rs

```diff
@@ -16,3 +18,22 @@ fn test_poll() {
     assert_eq!(nfds, 1);
     assert!(fds[0].revents().unwrap().contains(POLLIN));
 }
+
+#[cfg(any(target_os = "android", target_os = "dragonfly", target_os = "freebsd",
```

Comment:
It'd be great to have comments either inline or a big summary one that explains what this is doing and any gotchas that exist in this test. With a project as diverse as nix and the complexity of some of the APIs it exposes I think it's helpful to provide comments.